### DISCUSSION TITLE
Field conditions - how to use interface options

### DISCUSSION THREAD

**Discussion by H14** [directus/directus#11846] [Archive]
Hi

I can't seem to get the interface options from the field conditions to work, nor can I find what they are meant to do.
<img width="844" alt="image" src="https://user-images.githubusercontent.com/892420/155530583-a97b2fb9-3082-4ada-8ed8-454e5c4e44e5.png">

Wherever I access the field conditions, I'm always presented the above notices.
There is no way to select a collection, I have no idea what is misconfigured.

Is there something I need to configure or enable to use these features? Are they documented anywhere?

Many thanks

Hans

**Comment by benhaynes** [comment]
CC @erondpowell

**Reply by erondpowell** [reply]
Hey there @H14 

 >Wherever I access the field conditions, I'm always presented the above notices.

Can you tell me how/where in the app you accessed Interface Options?
Also, which version are you on?

**Reply by H14** [reply]
Settings > Data Model > (Pick a field) > Field options > Conditions (sidebar) > (make a new one or pick an existing one)

Version 9.5.2

Thanks!

**Reply by erondpowell** [reply]
Okay. **What kinds of Fields are you trying to set interface options on?**

I recreated a very similar situation with an O2M Field type:
<img width="832" alt="Screen Shot 2022-02-24 at 11 01 08 AM" src="https://user-images.githubusercontent.com/62306630/155571806-f3c14766-aef4-45aa-9e62-4025d9005ee7.png">

Here is some background info. Interface Options are different for each Field Type and some Field Types have no Interface Options at all. In general, the interface options let you set some fine-grain details about a Field. For example:

**_Interface Options for Input Fields:_**
<img width="824" alt="Screen Shot 2022-02-24 at 11 26 24 AM" src="https://user-images.githubusercontent.com/62306630/155575797-f94f6886-c6c6-4833-9124-3dfea386c6d3.png">

**_Interface options for Datetime Fields:_**
<img width="828" alt="Screen Shot 2022-02-24 at 11 24 42 AM" src="https://user-images.githubusercontent.com/62306630/155575498-e01d2c52-8c94-4a68-b88b-b2a6e7931953.png">

**Comment by hanszoons** [comment]
In this case I was trying to set it up for a M2O. I understand that the interface options depend on the field type.
But I don't understand how it is supposed to work in the context of a condition.

Interface options for the M2O field:
<img width="856" alt="image" src="https://user-images.githubusercontent.com/91126835/155676017-bbd98c76-6042-473a-847f-127cf9aac7b7.png">

Interface options for the same field, but from within a condition:
<img width="602" alt="image" src="https://user-images.githubusercontent.com/91126835/155676104-34387a11-5fe2-4301-887f-c97d64a721ff.png">

In your O2M example you have the same notices as me 'select a collection' but no way to select one, and the filter is unavailable because it is supposedly misconfigured. Does this mean that it is impossible to conditionally set interface options for these field types? Why would that be? Or is it a bug?

Thanks!

**Reply by azrikahar** [reply]
I believe it may be the same bug as mentioned here: https://github.com/directus/directus/discussions/10612#discussioncomment-1845360

**Comment by rasco** [comment]
No matter what I do, when I have a M2O collection and try to add a condition to it, on the bottom it always says "the collection field option is misconfigured".
I first created the relationship and saved it. I made sure it's working. It's looking good on the SQL side as well.
But when I go to the condition screen and edit/add conditions it always says it's misconfigured.

<img width="954" alt="Screenshot 2022-05-26 at 1 48 11 PM" src="https://user-images.githubusercontent.com/289536/170424889-439ec275-5ed1-4993-8b10-868f381bc75e.png">

Also another issue, might not be related: I can never get conditions with related objects to work, see screenshot:
<img width="959" alt="Screenshot 2022-05-26 at 1 52 33 PM" src="https://user-images.githubusercontent.com/289536/170425443-8bd54a20-e685-4e52-85b5-351cd7f11eb1.png">

**Reply by rasco** [reply]
Seems that it doesn't matter what type of relationship field it is. 
It always says "misconfigured"

- PostgreSQL
- all relations seem to be working normally otherwise
- it also happens on two freshly created collections

**Comment by tmaihoff** [comment]
any update here? I'm trying to configure a condition as well but it says "The collection field option is misconfigured".

EDIT: In my case was trying to set up a condition to only allow the selection of subcateries belonging to the selected main category. I was able to solve this via the "Interface" option instead https://github.com/directus/directus/discussions/7071#discussioncomment-5240868

**Comment by drupalshift** [comment]
Same here I'm trying to create a M2M field and set a filter based on a condition but i get The collection field option is misconfigured

**Comment by jagyas** [comment]
Same here for M2M
